[WIP] Fix #1722: Added transition css class to appropriate elements

bkim49 opened the pull request:
Added transition css to appropriate elements in userbar.less and editor.less to change transition time effectively. 

At the moment, the current transition time for both files are set to 0.15s. I can change the time back to 0.5s or whatever you think is the best transition time @flukeout @gideonthomas.

flukeout commented:
Checking now!

flukeout commented on public/editor/stylesheets/editor.less:
```diff
@@ -258,6 +260,7 @@ body {
     width: 15px;
     position: relative;
     top: 0;
+	.transition;
```
Please keep spacing / indentation consistent. I think you're using Tabs while the source file is using spaces.

flukeout commented:
Looks good overall! Couple of things: 

* Please fix the spacing / indentation (commented on one of the lines where this happens)
* Please add an ``ease-out`` timing to the transitions.

I think you can add it to the each of the ``.transition {`` definitions in ``userbar.less`` and ``editor.less`` - currently there is no timing function defined.

Thanks!

bkim49 commented:
Sorry for late reply!

I have changed the spacing and added transition timing function as well. I tried to look at the difference of having transition timing function as ease out or ease in etc, but I could not really tell a  difference. 

@flukeout could you let me know if the changes were correct?

flukeout commented:
Will review a little later today, thank you!

Pomax commented on public/editor/stylesheets/editor.less:
```diff
@@ -850,13 +850,14 @@ body {
+    transition: @transitionTime ease-out;
     -webkit-backface-visibility: hidden;
     backface-visibility: hidden;
+	transition-timing-function: ease-out;
```
looks like indentation went wrong here again

gideonthomas replied:
@bkim49 I think you're using tabs vs. spaces. We use 2-space indents throughout the code so would you be able to convert this to 4 spaces?

flukeout commented on public/resources/stylesheets/userbar.less:
```diff
@@ -255,19 +258,21 @@ body {
   -ms-transition: opacity @transitionTime ease-in-out;
   -o-transition: opacity @transitionTime ease-in-out;
   transition: opacity @transitionTime ease-in-out;
+  transition-timing-function: opacity @transitionTime ease-out;
```
Was this line added or changed by mistake? It's incorrect.

bkim49 commented:
@Pomax thank you for letting me know!

@flukeout omg. I was testing stuff and forgot to remove them. I should really double check my work. Ill fix this as well. Thank you.

humphd commented:
@bkim49 np, we all leave things in our commits.  One tip: when you go to submit a PR, always go and read the `Files changed` tab to see what's being sent, in case you missed something.

bkim49 commented:
@humphd I will remeber that. Thank you.

flukeout commented:
@bkim49 Just want to check in on this PR - will you be updating it with the minor fixes?

bkim49 commented:
@flukeout I have fixed the minor error on my part. Sorry for taking so long for this fix!

humphd commented:
@bkim49 looks like this will need a rebase for changes in `public/editor/stylesheets/editor.less`.  Have you made all the other changes?

bkim49 commented:
@flukeout @Pomax I have made changes to the requests!

flukeout commented:
Hi @bkim49 not sure if you noticed but the indentation still needs to be fixed on ``line 872`` in ``editor.less`` - you can see the comments if you click on the  Files Changed tab above and scroll down to see the comments. Almost there!

flukeout commented:
Ok - testing it out again!

flukeout commented:
Thanks for your work on this @bkim49 !
